src/parse.rs

```diff
@@ -351,6 +351,18 @@ impl<'a> Parser<'a> {
             (end, Expr::Assertion(Assertion::StartText))
         } else if b == b'z' && !in_class {
             (end, Expr::Assertion(Assertion::EndText))
+        } else if b == b'Z' && !in_class {
+            (
+                end,
+                Expr::LookAround(
+                    Box::new(Expr::Delegate {
+                        inner: "\n*$".to_string(),
+                        size: 1,
+                        casei: self.flag(FLAG_CASEI),
```

Comment:
I think `size` should be 0, no? It's a zero-length match. (I don't recall how size for delegates is actually used, but yeah.)

Also, I think `casei` should just be `false` here, regardless of the flag. (`\n*$` never needs to be wrapped in `(?i:`.)

Reply:
Fixed. Also you were write about `size` - [it is used to determine `min_size`,](https://github.com/forkeith/fancy-regex/blob/c1b8a315239f07a2dc55dd1c5f0f22f64a5db03b/src/analyze.rs#L149) so 0 is indeed the correct size to use here.